pkg/checkup/workspace.go

```diff
@@ -0,0 +1,346 @@
+		RoleRef:    rbacv1.RoleRef{APIGroup: rbacv1.GroupName, Kind: ClusterRoleKind, Name: clusterRole.Name},
+	}
+}
+
+func createEnvVarsFromSpec(parametersMap map[string]string) []corev1.EnvVar {
+	var checkupEnvVars []corev1.EnvVar
+	if len(parametersMap) > 0 {
+		for k, v := range parametersMap {
+			if k != "" && v != "" {
+				checkupEnvVars = append(checkupEnvVars, corev1.EnvVar{Name: strings.ToUpper(k), Value: v})
+			}
+		}
+	}
+	return checkupEnvVars
+}
+
+func newCheckupJob(image string, envs []corev1.EnvVar, resources corev1.ResourceList, name string, timeout time.Duration) *batchv1.Job {
+	checkupContainer := newCheckupJobContainer(image, envs, resources)
+	terminationGracePeriodSeconds := int64(5)
+	checkupPodSpec := newPodTemplateSpec(&terminationGracePeriodSeconds, []corev1.Container{checkupContainer})
+	activeDeadlineSeconds := int64(timeout.Seconds())
+	backoffLimit := int32(0)
+	ttlSecondsAfterFinished := int32(5)
+	return &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{Name: name},
+		Spec: batchv1.JobSpec{
+			ActiveDeadlineSeconds:   &activeDeadlineSeconds,
+			BackoffLimit:            &backoffLimit,
+			TTLSecondsAfterFinished: &ttlSecondsAfterFinished,
+			Template:                checkupPodSpec,
+		},
+	}
+}
+
+func newCheckupJobContainer(image string, envs []corev1.EnvVar, resources corev1.ResourceList) corev1.Container {
+	return corev1.Container{
+		Name:  "checkup",
+		Image: image,
+		Env:   envs,
+		Resources: corev1.ResourceRequirements{
+			Limits:   resources,
+			Requests: resources,
+		},
+		ImagePullPolicy: corev1.PullAlways,
+	}
+}
+
+func newPodTemplateSpec(terminationGracePeriodSeconds *int64, containers []corev1.Container) corev1.PodTemplateSpec {
+	return corev1.PodTemplateSpec{
+		ObjectMeta: metav1.ObjectMeta{},
+		Spec: corev1.PodSpec{
+			ServiceAccountName:            serviceAccountName,
+			RestartPolicy:                 corev1.RestartPolicyNever,
+			TerminationGracePeriodSeconds: terminationGracePeriodSeconds,
+			Containers:                    containers,
+		},
+	}
+}
+
+func (w *workspace) SetupCheckupWorkspace(client *kubernetes.Clientset) error {
+	const errMsgPrefix = "failed to create checkup environment"
+
+	if err := w.createNamespace(client); err != nil {
+		return fmt.Errorf("%s: %v", errMsgPrefix, err)
+	}
+
+	if err := w.createServiceAccount(client); err != nil {
+		return fmt.Errorf("%s: %v", errMsgPrefix, err)
+	}
+
+	if err := w.createResultsConfigMap(client); err != nil {
+		return fmt.Errorf("%s: %v", errMsgPrefix, err)
+	}
+
+	if err := w.createResultsConfigMapWriterRole(client); err != nil {
+		return fmt.Errorf("%s: %v", errMsgPrefix, err)
+	}
+
+	if err := w.createRoleBindings(client); err != nil {
+		return fmt.Errorf("%s: %v", errMsgPrefix, err)
+	}
+
+	if err := w.createClusterRoleBindings(client); err != nil {
+		return fmt.Errorf("%s: %v", errMsgPrefix, err)
+	}
+
+	return nil
+}
+
+func (w *workspace) createNamespace(client *kubernetes.Clientset) error {
+	var err error
```

Comment:
I did try at at first but it didn't compiled:
`w.namespace, err := client.CoreV1().Namespaces().Create(context.Background(), w.namespace, metav1.CreateOptions{})`
Wont pass compilation since `w.namespace` already defined, if this is what meant.
Next time it would be easier if you would provide an example :slightly_smiling_face: